simple-dlna-browser: Add new package

LoveIsGrief opened the pull request:
###### Description of changes

Add a  useful CLI script I found that's way light than VLC and other GUI apps since SSDP is pretty much a text-based protocol built on top of HTTP. It has no license attached so I marked it as unfree.
If the author adds a license, we will be able to specify it. @javier-lopez, would you be willing to add a license to your `simple-dlna-browser` script for its inclusion in NixOS?

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

javier-lopez commented:
Hello @LoveIsGrief , sure, would it be enough to add a header at the top of the script?

javier-lopez commented:
I've licenced the script with the https://en.wikipedia.org/wiki/Fair_License , I hope this makes clear and easy the inclusion of the file, happy new year 🍻 , https://github.com/javier-lopez/learn/blob/master/sh/tools/simple-dlna-browser#L9

LoveIsGrief commented:
Thank you @javier-lopez !

Commit:
lib/licenses add Fair license

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/1681

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -8352,6 +8352,13 @@
     githubId = 265084;
     name = "Jason O'Conal";
   };
+  loveisgrief = {
```
Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

LoveIsGrief replied:
Done in 7df27f1

SuperSandro2000 commented on pkgs/tools/networking/simple-dlna-browser/default.nix:
```diff
@@ -0,0 +1,57 @@
+in
+pkgs.stdenv.mkDerivation {
+  name = "simple-dlna-browser";
+  meta = {
```
meta must be last

LoveIsGrief replied:
Done in 9ca8ba8

SuperSandro2000 commented on pkgs/tools/networking/simple-dlna-browser/default.nix:
```diff
@@ -0,0 +1,57 @@
+  paths = builtins.concatStringsSep ":" binPaths;
+in
+pkgs.stdenv.mkDerivation {
+  name = "simple-dlna-browser";
```
Please use pname+version

LoveIsGrief replied:
There is no version. Should I take the commit ID as the version?

SuperSandro2000 replied:
Please take the date of the commit like https://github.com/NixOS/nixpkgs/blob/ee96940bad52677b362baad3150a0931abb74350/pkgs/applications/audio/adlplug/default.nix#L49

LoveIsGrief replied:
Done in 
0f356ab

SuperSandro2000 commented on pkgs/tools/networking/simple-dlna-browser/default.nix:
```diff
@@ -0,0 +1,57 @@
+    maintainers = with lib.maintainers; [ loveisgrief ];
+  };
+
+  src = pkgs.fetchgit {
```
fetchFromGitHub must be used

LoveIsGrief replied:
Done in 9ca8ba8

SuperSandro2000 commented on pkgs/tools/networking/simple-dlna-browser/default.nix:
```diff
@@ -0,0 +1,57 @@
+    sha256 = "sha256-jwPAEgti5DrjNnNVW/61GGTP+QedGAjz9kPwYhMAdTk=";
+  };
+
+  buildInputs = buildInputs;
```
Please inline

LoveIsGrief replied:
Done in 9ca8ba8

SuperSandro2000 commented on pkgs/tools/networking/simple-dlna-browser/default.nix:
```diff
@@ -0,0 +1,57 @@
+    # Use a wrapper script to prefix PATH with the buildInput
+    # This is easier than trying to modify the script with sed
+    mkdir -p $out/bin
+    echo '#!/usr/bin/env bash
+    export PATH="${paths}:''${PATH:-}"' > $outputWrapper
+    echo exec $out/bin/wrapped-simple-dlna-browser '"''${@}"' >> $outputWrapper
```
Use makeWrapper

LoveIsGrief replied:
Done in 9ca8ba8

SuperSandro2000 commented on pkgs/tools/networking/simple-dlna-browser/default.nix:
```diff
@@ -0,0 +1,57 @@
+    gnused
+    socat
+  ];
+  binPaths = builtins.map (x: "${x}/bin") buildInputs;
```
Use lib.makeBInPath

LoveIsGrief replied:
Done in 9ca8ba8

SuperSandro2000 reviewed (changes requested):
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

Commit:
maintainers: Add loveisgrief

Commit:
simple-dlna-browser: Add new package

A useful CLI script I found that's way light than VLC and other GUI apps since SSDP
 is pretty much a text-based protocol built on top of HTTP.

LoveIsGrief commented:
Thanks for the review @SuperSandro2000 !

SuperSandro2000 commented on pkgs/tools/networking/simple-dlna-browser/default.nix:
```diff
@@ -0,0 +1,54 @@
+{ pkgs, lib, makeWrapper, ...}:
```
pkgs should not be used as an input in normal packages. Also ... is not necessary here.

SuperSandro2000 commented on pkgs/tools/networking/simple-dlna-browser/default.nix:
```diff
@@ -0,0 +1,54 @@
+    sha256 = "sha256-jwPAEgti5DrjNnNVW/61GGTP+QedGAjz9kPwYhMAdTk=";
+  };
+
+  inherit (buildInputs);
```
Are those packages used at build time?

LoveIsGrief replied:
`sed` is used in the install phase and `socat` is necessary for the script's full functionality.

LoveIsGrief replied:
Found a few more dependencies in the script and added them to the args. Nothing is really built, so I could remove the line entirely in 268feff

SuperSandro2000 commented on pkgs/tools/networking/simple-dlna-browser/default.nix:
```diff
@@ -0,0 +1,54 @@
+    mkdir -p $out/bin
+    cp sh/tools/simple-dlna-browser $outputWrapped
+    makeWrapper $outputWrapped $outputWrapper \
+      --prefix PATH : ${lib.makeBinPath buildInputs}
```
If they are only used for wrapping here then buildInputs above is doing nothing and we can simplify this to 

```suggestion
      --prefix PATH : ${lib.makeBinPath [ gnused socat ]}
```

LoveIsGrief replied:
I just tried this, but got `error: undefined variable 'gnused'`. How does one access these packages without using `pkgs` ?

LoveIsGrief replied:
Ah... I get it now. `callPackage` uses dependency injection. We desperately need documentation on the pattern as well as to generate documentation from nix code like `javadoc` and `cargo doc`...

Anyway, this should work now in 268feff

SuperSandro2000 commented on pkgs/tools/networking/simple-dlna-browser/default.nix:
```diff
@@ -0,0 +1,54 @@
+  meta = {
+    description = "Query dlna media servers";
+    longDescription = ''
+      A bash script that allows simple interactions with [DLNA] media servers.
+      It is a light-weight, CLI alternative to [VLC], [Kodi], and other players.
+
+      [DLNA]: https://en.wikipedia.org/wiki/Digital_Living_Network_Alliance
+      [Kodi]: https://kodi.tv/
+      [VLC]: https://www.videolan.org/vlc/
```
I am not sure if the longDescription is actual markdown. It could also be that it is just written as is into the result.

LoveIsGrief replied:
The [documentation](https://nixos.org/manual/nixpkgs/stable/#var-meta-longDescription) says it's CommonMark.

> An arbitrarily long description of the package in [CommonMark](https://commonmark.org/) Markdown.

SuperSandro2000 reviewed (commented):
I missed some things the first time, sorry.

LoveIsGrief commented:
That's alright, I learned something new. Thanks again
